akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/Running.scala

```diff
@@ -484,10 +488,9 @@ private[akka] object Running {
     }
 
     signal match {
-      case Some(sig) =>
-        setup.onSignal(state, sig, catchAndLog = false)
+      case Some(sig) if setup.onSignal(state, sig, catchAndLog = false) =>
```

Comment:
I don't think we should perform side effect in case guard
```
  case Some(sig)
    if (setup.onSignal(state, sig, catchAndLog = false)) Behaviors.same else Behaviors.unhandled
  case None =>
    Behaviors.unhandled // unexpected journal response
```